libs/zedUpload/awsutil/s3sync.go

```diff
@@ -245,17 +245,19 @@ func getNeededParts(cWriterOptions *writerOptions, bname, bkey string, doneParts
 }
 
 func (s *S3ctx) DownloadFile(fname, bname, bkey string,
-	bsize int64, doneParts types.DownloadedParts, prgNotify NotifChan) (types.DownloadedParts, error) {
+	objMaxSize int64, doneParts types.DownloadedParts, prgNotify NotifChan) (types.DownloadedParts, error) {
 
 	var fd *os.File
-	var err error
 	var wg sync.WaitGroup
 
-	if bsize == 0 {
-		err, bsize = s.GetObjectSize(bname, bkey)
-		if err != nil {
-			return doneParts, err
-		}
+	err, bsize := s.GetObjectSize(bname, bkey)
+	if err != nil {
+		return doneParts, err
+	}
+
+	if objMaxSize != 0 && bsize > objMaxSize {
+		return types.DownloadedParts{PartSize: S3PartSize},
+			fmt.Errorf("configured image size (%d) is less than blob size (%d)", objMaxSize, bsize)
```

Comment:
is "blob size" a correct wording here? Would it make sense rather say something like "Configured volume size is smaller the the size of image associated with it"?